plugins/entity_filter.py

```diff
@@ -334,13 +401,18 @@ def entity_filter_task(self, db_id: int) -> str:
    
     if not attributes: # empty string or None
         attributes = []
-        
+    
+    # number of rows to be sampled
+    n_sampled_rows : int = 0
     if n_rows is None and not id_list:
-        raise ValueError("Row number argument not provided!")
+        # only filter columns
+        id_list = {}
```

Comment:
Empty `{}` always result in a dict. An empty set must use `set()` constructor. Also this would be better if id_list is only used for the id list and another name for the set (`id_set`) to clearly differentiate the types. The type of id_list is also wrong (is `Optional[str]` but should be `Optional[List[str]]`).
The whole conversion can also be streamlined:

```python
id_set: Set[str] = set(params.get("id_list", []))
```